File: code/sdk/android/cloudsheet-Android-sdk/src/main/kotlin/com/opencloudsheet/model/file/ICloudFile.kt
package com.opencloudsheet.model.file

/**
 * Generic interface representing a file or folder in cloud storage.
 * This abstraction works across different cloud providers (OneDrive, Google Drive, Dropbox, etc.)
 */
interface ICloudFile {
    fun getId(): String
    fun getName(): String
    fun isFolder(): Boolean
    fun getPath(): String?
    fun getSize(): Long?
}

File: code/sdk/ios/OpenCloudSheet/Sources/OpenCloudSheet/Model/File/ICloudFile.swift
//
//  ICloudFile.swift
//  OpenCloudSheet
//
//  Generic interface representing a file or folder in cloud storage
//

import Foundation

protocol ICloudFile {
    func getId() -> String
    func getName() -> String
    func isFolder() -> Bool
    func getPath() -> String?
    func getSize() -> Int64?
}
